client/internal/packagessyncer.go

```diff
@@ -49,6 +53,9 @@ func (s *packagesSyncer) Sync(ctx context.Context) error {
 	}()
 
 	// Prepare package statuses.
+	// Grab a lock to make sure that package statuses are not overriden by
+	// another call to Sync running in parallel.
+	s.mut.Lock()
```

Comment:
if either s.initStatuses() or s.clientSyncedState.SetPackageStatuses() returns an error, the function will return without unlocking the mutex

Reply:
You're right thanks for the catch. We can Unlock in case we return with an error here.

Reply:
FYI @srikanthccv I also pushed a test in 2048d4de8c4be50653ec1dd5edef2fc3c9d996bc to showcase that unlocking works properly in case of an early error in `s.initStatuses()`.

There's not really a way to error out on `SetPackageStatuses()` right now since the only failure mode is covered by initStatuses so I hope it's ok if it's not expliticly covered by another test.